client/signup/main.jsx

```diff
@@ -504,45 +504,39 @@ class Signup extends React.Component {
 		return window.location.origin + path;
 	};
 
-	// `flowName` is an optional parameter used to redirect to another flow, i.e., from `main`
-	// to `ecommerce`. If not specified, the current flow (`this.props.flowName`) continues.
-	goToStep = ( stepName, stepSectionName, flowName = this.props.flowName ) => {
-		if ( this.state.scrolling ) {
-			return;
-		}
-
-		// animate the scroll position to the top
-		const scrollPromise = new Promise( ( resolve ) => {
-			this.setState( { scrolling: true } );
-
-			const ANIMATION_LENGTH_MS = 200;
-			const startTime = window.performance.now();
-			const scrollHeight = window.pageYOffset;
-
-			const scrollToTop = ( timestamp ) => {
-				const progress = timestamp - startTime;
-
-				if ( progress < ANIMATION_LENGTH_MS ) {
-					window.scrollTo( 0, scrollHeight - ( scrollHeight * progress ) / ANIMATION_LENGTH_MS );
-					window.requestAnimationFrame( scrollToTop );
-				} else {
-					this.setState( { scrolling: false } );
+	// animate the scroll position to the top
+	scrollToTop() {
+		return new Promise( ( resolve ) => {
+			const resolveScrolling = () => {
+				if ( 0 === window.pageYOffset ) {
+					window.removeEventListener( 'scroll', resolveScrolling );
 					resolve();
 				}
 			};
 
-			window.requestAnimationFrame( scrollToTop );
+			window.addEventListener( 'scroll', resolveScrolling );
```

Comment:
Not a big fan of having to do this, but not sure how else to achieve it natively. `window.scrollTo` is non-blocking. 🤔
Not sure if the callback ought to be throttled either - looks like a candidate.